File: 2023/06/part_two.js
// @ts-check

import fs from "fs";

const data = fs.readFileSync("puzzle_input.txt", "utf8");
const lines = data.split(/\r?\s*\n/u).filter((line) => line !== "");
const linesLength = lines.length;

const [timePrefix, timeDataString] = lines[0].split(":");
const [distPrefix, distDataString] = lines[1].split(":");

const timeToRace = parseInt(timeDataString.trim().split(/\s+/u).join(""));
const recordDistance = parseInt(distDataString.trim().split(/\s+/u).join(""));

const a = -1.0;
const b = timeToRace;
const c = recordDistance * -1.0;

const discriminant = Math.sqrt(b ** 2 - 4 * a * c);
const root1 = Math.ceil((-b + discriminant) / (2 * a));
const root2 = Math.floor((-b - discriminant) / (2 * a));
const possibilities = root2 - root1 + 1;

console.log(possibilities);

File: 2023_py/06/part_two.py
import math as m

lines = [line.strip() for line in open("puzzle_input.txt")]
time_info = lines[0].split(":")[1].strip()
dist_info = lines[1].split(":")[1].strip()

a = -1
b = int("".join([info.strip() for info in time_info]))
c = int("".join([info.strip() for info in dist_info])) * -1

discriminant = m.sqrt(b**2 - 4 * a * c)
root_1 = m.ceil((-b + discriminant) / (2 * a))
root_2 = (-b - discriminant) // (2 * a)

print(root_2 - root_1 + 1)
